Bottom: 3736927f2f621ae83d4830280ac3a3ab93274c06
Top:    baba7bef11692a375087b1953d6066dbe591e4e2
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-07-21 22:36:59 +0800

Refresh of small

---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index e3d7ea1..f2a22b1 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -628,7 +628,8 @@ static int vhost_net_rx_peek_head_len(struct vhost_net *net, struct sock *sk)
 		endtime = busy_clock() + vq->busyloop_timeout;
 
 		while (vhost_can_busy_poll(&net->dev, endtime) &&
-		       !sk_has_rx_data(sk) &&
+		       (!sk_has_rx_data(sk) ||
+			vhost_vq_avail_empty(&net->dev, &rvq->vq)) &&
 		       vhost_vq_avail_empty(&net->dev, vq))
 			cpu_relax();
